mev_inspect/schemas/classified_traces.py

```diff
@@ -24,32 +24,53 @@ class Protocol(Enum):
     zerox = "0x"
 
 
-class ClassifiedTrace(BaseModel):
+class ClassifiedTrace(Trace):
+
     transaction_hash: str
     block_number: int
-    trace_type: TraceType
     trace_address: List[int]
     classification: Classification
-    protocol: Optional[Protocol]
-    abi_name: Optional[str]
-    function_name: Optional[str]
-    function_signature: Optional[str]
-    inputs: Optional[Dict[str, Any]]
+
+    # Optional
+    error: Optional[str]
     to_address: Optional[str]
     from_address: Optional[str]
     gas: Optional[int]
     value: Optional[int]
     gas_used: Optional[int]
-    error: Optional[str]
+    protocol: Optional[Protocol]
+    function_name: Optional[str]
+    function_signature: Optional[str]
+    inputs: Optional[Dict[str, Any]]
+    abi_name: Optional[str]
 
     class Config:
+        validate_assignment = True
         json_encoders = {
             # a little lazy but fine for now
             # this is used for bytes value inputs
             bytes: lambda b: b.hex(),
         }
 
 
+class CallTrace(ClassifiedTrace):
+
+    to_address: str
+    from_address: str
+
+
+class DecodedCallTrace(CallTrace):
+    inputs: Dict[str, Any]
+    abi_name: str
+
+    # Optional
```

Comment:
I think remove this comment

Controversial opinion, but in general I'm not a fan of comments because they're out of sync with the code itself (ppl can change code without remembering to change comments => more confusion / issues - do I trust the comment or my own interpretation of the code?). Better to just write clear code.

Example for this one:
- This comment is here
- Someone doesn't see the comment and puts an optional in the top ones (now the comment is wrong)
- Someone else sees the comment and writes their code assuming that those are the only Optional arguments
- Later they get an error in their code because they missed one someone put up top

I think just slotting them in the bottom is enough to imply it without making some guarantee